plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/AuthorizationCodeListener.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import org.eclipse.swt.browser.TitleEvent;
+import org.eclipse.swt.browser.TitleListener;
+
+/**
+ * Listens for title changes of HTML pages for {@link GoogleLoginBrowser} to capture an authorization
+ * code. (The authorization code after successful Google login will be given in the HTML title.)
+ */
+class AuthorizationCodeListener implements TitleListener {
+
+  private static final String SUCCESS_CODE_PREFIX = "Success code="; //$NON-NLS-1$
+
+  private GoogleLoginBrowser loginBrowser;
+
+  AuthorizationCodeListener(GoogleLoginBrowser loginBrowser) {
+    this.loginBrowser = loginBrowser;
+  }
+
+  @Override
+  public void changed(TitleEvent event) {
+    if (event.title.startsWith(SUCCESS_CODE_PREFIX)) {
```

Comment:
This code is from GPE, so I think it's ok. But I added a null check anyway to be safe, as I couldn't find a doc saying this is never null.